CIP-0118/README-nested.md

```diff
@@ -0,0 +1,556 @@
+values, respectively. Then
+
+extra assets/currency `:= { aid ↦ v | aid ↦ v ∈ consumed - produced , v > 0 }`
+
+missing assets/currency `:= { aid ↦ v | aid ↦ v ∈ produced - consumed , v > 0 }`
+
+### Transaction structure changes
+
+We add the following new fields to `TxBody` : 
+
+1. `requireBatchObservers : ℙ ScriptHash`
+- set of scripts that must be run by the top-level transaction in the batch (where `TxInfo` includes info of all batch transactions)
+
+2. `spendOuts : Ix ⇀ TxOut`
+- (indexed) outputs being spent 
+
+3. `corInputs : ℙ TxIn`
+- inputs corresponding to `spentOuts`
+
+4. `subTxIds : ℙ TxId`
+-  IDs of sub-transactions in the batch
+
+**Decision required** It could be better to include the sub-transactions in the body directly, or possibly in the script integrity hash
+
+**Decision required** Should this CIP additionally include the following :
+
+- `RdmrPtr ⇀ Redeemer` structure inside the body
+- datums inside the body
+- `requiredObservers` scripts that are checked by a transaction but are not for the purpose of validating a specific action
+
+The following field is added to `Tx`:
+
+1.  `subTxs : List Tx` 
+- list of sub-transactions
+
+### Plutus
+
+#### New version
+
+A new Plutus version, `PlutusV4`, will be required, since the constraints on what it means for 
+a transaction to be valid, as well as the tx structure itself, are changed. Transactions 
+that are top-level transactions with no sub-transactions are still able to interact 
+with V3 scripts. Transactions that are underspecified and/or using any new features 
+introduced here are not. 
+
+#### New script purposes
+
+The following two new script purposes are introduced : 
+
+1. `SpendOut : Ix → ScriptPurpose`, where the index indicates that a script is being run to validate spending a specific 
+output in `spendOuts`.
+
+2. `BatchObs : ScriptHash → ScriptPurpose`, where the script has indicates the script that must be run by the top-level 
+transaction to observe the validity of the batch.
+
+`ScriptsNeeded` and `getDatum` are modified accordingly to include scripts for these purposes. `collectPhaseTwoScriptInputs` 
+also requires an additional parameter for constructing `TxInfo` to either contain lists of `TxInfo` for sub-transactions or not.
+
+#### TxInfo
+
+The `TxInfo` shown to a PlutusV4 script has an added field :
+
+- `otherInfos : List TxInfo`, which is populated with the `TxInfo` data for all sub-transactions in the batch 
+whenever (1) the transaction for which info is being constructed is top-level, and (2) the script purpose for 
+which this info is constructed is `BatchObs`
+
+
+### Ledger Transition Changes
+
+The prototype for the changes can be found
+[here](https://github.com/IntersectMBO/formal-ledger-specifications/compare/polina-nested-txs)
+
+The transition structure is changed from 
+
+`LEDGERS -> LEDGER -> UTXOW -> UTXO -> UTXOS` 
+
+to 
+
+`LEDGERS -> LEDGER -> SWAPS -> SWAP -> UTXOW -> UTXO -> UTXOS`
```

Comment:
I think this will be better than the latter below, since sub transactions are morally part of a transaction and `LEDGER` historically was a rule that validated a single transaction.